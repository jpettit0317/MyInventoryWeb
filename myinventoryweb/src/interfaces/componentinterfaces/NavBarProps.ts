interface NavBarProps {
    navBarTitle: string
}

export default NavBarProps;