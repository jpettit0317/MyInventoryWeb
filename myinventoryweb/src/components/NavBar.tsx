import React from "react";
import { AppBar, Toolbar, Typography, Button, IconButton} from "@material-ui/core";
import useStyles from "../componentstyles/navbarstyles";
import NavBarProps from "../interfaces/componentinterfaces/NavBarProps";

function NavBar({navBarTitle = "MyInventory"}: NavBarProps) {
    const navBarStyles = useStyles();

    return (
        <div className={navBarStyles.root}>
            <AppBar position="static" className={navBarStyles.background}>
                <Toolbar>
                    <Typography variant="h6" className={navBarStyles.title}>
                       {navBarTitle} 
                    </Typography>
                </Toolbar>
            </AppBar>
        </div>
    );
}

export default NavBar;