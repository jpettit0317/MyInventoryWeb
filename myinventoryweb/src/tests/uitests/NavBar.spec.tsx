import React from "react";
import renderer from "react-test-renderer";
import NavBar from "../../components/NavBar";
import NavBarProps from "../../interfaces/componentinterfaces/NavBarProps";

describe('<Nav Bar />', () => {
    class DummyNavBarProps implements NavBarProps {
        readonly navBarTitle: string = "";

        constructor(newTitle: string = "") {
            this.navBarTitle = newTitle;
        }
    }

    const navBarProps = new DummyNavBarProps("TestNavBar");

    it('title should be set to TestNavBar', () => {
        const navBarComponentTree = renderer.create(
            <NavBar navBarTitle={ navBarProps.navBarTitle } />
        ).toJSON();

        expect(navBarComponentTree).toMatchSnapshot();
    });
});