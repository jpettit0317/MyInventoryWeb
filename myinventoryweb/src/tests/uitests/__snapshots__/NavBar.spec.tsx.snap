// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav Bar /> title should be set to TestNavBar 1`] = `
<div
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-background-4 MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <h6
        className="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
      >
        TestNavBar
      </h6>
    </div>
  </header>
</div>
`;
